stable/nginx-ingress/templates/controller-daemonset.yaml

```diff
@@ -28,6 +28,9 @@ spec:
           args:
             - /nginx-ingress-controller
             - --default-backend-service={{ if .Values.defaultBackend.enabled }}{{ .Release.Namespace }}/{{ template "defaultBackend.fullname" . }}{{ else }}{{ .Values.controller.defaultBackendService }}{{ end }}
+          {{- if .Values.controller.publishService.enabled }}
```

Comment:
Since this flag is new in 0.9 (still in beta), let's add that conditional. This should work: `{{- if and (contains "0.9" .Values.controller.image.tag) .Values.controller.publishService.enabled }}`